[AGM-1813] Create 'Solving Problems' section on AdGuard for Safari (#416)
AdguardTeam/KnowledgeBase: docs/adguard-for-safari/solving-problems/rule-limit.md

---
title: Rule limit issues
sidebar_position: 1
---

:::info

> This article is about AdGuard for Safari, which safeguards only your Safari browser. To protect your entire device, [download the AdGuard app](https://agrd.io/download-kb-adblock)

Reviewer: > safeguards 

вроде ж техническая статья. Как это по-русски будет?)

Author: хз, этот текст до меня был, это общая плашка для кб Af for Safari

Author: @ss-orlova помоги плиз) окей так или поправим на works какой-нить

Reviewer 2: ну вроде все прилично

:::

## Safari content blocker rule limit

> AdGuard for Safari uses the Content Blocking API to apply filtering rules to pages in the Safari browser. Currently, a maximum of 150,000 rules are allowed for each content blocking extension.

Reviewer 2: вот тут еще остался дефис и вроде все!

> Safari Content Blocker functionality is an indispensable tool for ad-blocking in Safari: it gives a way to apply filtering rules to pages accessed in Safari. Currently, as stated in the Safari Content Blocking API documentation, up to 150k rules is allowed in one content blocker.

Reviewer: Я бы чуть покороче написал первое предложение
> Safari Content Blocker API is used to apply filtering rules to pages in the Safari browser.

Author: @ss-orlova что думаешь) мошт есть индейка

Reviewer 2: звучит как хорошее предложение)

> Safari Content Blocker functionality is an indispensable tool for ad-blocking in Safari: it gives a way to apply filtering rules to pages accessed in Safari. Currently, as stated in the Safari Content Blocking API documentation, up to 150k rules is allowed in one content blocker.

Reviewer 2: Safari Content Blocker functionality → the Safari Content Blocker functionality
ad-blocking → ad blocking
it gives a way → it allows AdGuard for Safari to
150k → 150K или 150,000

но лучше действительно подсократить

```suggestion
AdGuard for Safari uses the Content Blocking API to apply filtering rules to pages in the Safari browser. Currently, a maximum of 150,000 rules are allowed for each content blocking extension.
```

Reviewer: строчная `k` - корректно, `M` для миллионов (в стате на доиде тоже неверно)
https://en.wikipedia.org/wiki/Kilo-

AdGuard for Safari has 6 content blockers:

- AdGuard General
- AdGuard Privacy
- AdGuard Social
- AdGuard Security
- AdGuard Other
- AdGuard Custom

> That means user can use up to 900k filtering rules total.

Reviewer 2: ```suggestion
This makes a total of 900,000 filtering rules.
```

> However, **you can only have up to 150k in one content blocker**. If you exceed that limit, part of the rules will not be applied and that could lead to some things not being blocked.

Reviewer 2: ```suggestion
However, **an extension can contain up to 150,000 rules**. If you exceed this limit, some of the rules will not be applied. This may lead to incorrect blocking.
```

> ## How to check that if you are exceeding rule limit

Reviewer 2: ```suggestion
## How to check if you are exceeding the rule limit
```

>  1. Open AdGuard for Safari settings by clicking its icon beside the Safari's search bar.

Reviewer 2: ```suggestion
 1. Click the AdGuard for Safari icon next to the search bar and open its settings.
```

Reviewer: Это не единственный способ открытия. R тому же икона сверху может быть выключена настройками приложения, или просто не поместиться при переполнении панели.

Author: какой еще способ? просто как любую прилагу, например из списка прилаг?

Author: поменял везде на просто "open AdGuard for Safari app"

 2. Scroll down and click *Content blockers*.

>  3. Under each content blocker, you can see the number of enabled rules and enabled filters.

Reviewer 2: иначе кажется что там количество включенных фильтров
```suggestion
 3. Below each content blocker, you can see the enabled filters and the number of active rules.
```

    ![Content blockers](https://cdn.adtidy.org/content/Kb/ad_blocker/safari/adg-safari-cb.png)

>  If a content blocker has more than 150,000 rules, go to *Filters* tab, select a filter category and disable filters or user rules you don't need.

Reviewer: Корявый апостроф вырвиглазно выглядит, по крайней мере в документациях. Как и витиеватые кавычки в других местах.

>  If a content blocker has more than 150,000 rules, go to *Filters* tab, select a filter category and disable filters or user rules you don't need.

Reviewer: @vyanovsky предложенный вариант норм, только апостроф нормальный оставь, не макосный. В технических статьях `'` используют (у нас надобавляли зачем-то `’`).

>  If a content blocker has more than 150,000 rules, go to *Filters* tab, select a filter category and disable filters or user rules you don't need.

Reviewer 2: ну потому что мы используем этот апостроф. в остальных статьях тоже потихоньку сделаем консистентно
